tidy3d/components/microwave/mode_spec.py

```diff
@@ -16,10 +17,107 @@
     ImpedanceSpecType,
 )
 from tidy3d.components.mode_spec import AbstractModeSpec
-from tidy3d.components.types import annotate_type
+from tidy3d.components.types import Coordinate2D, annotate_type
+from tidy3d.components.types.base import ArrayFloat2D
 from tidy3d.constants import fp_eps
 from tidy3d.exceptions import SetupError
 
+# Threshold for determining whether a mode is QuasiTEM
+DEFAULT_QUASI_TEM_THRESHOLD = 0.97
+
+ConductorIdentifierType = Union[Coordinate2D, str, ArrayFloat2D]
+
+
+class TerminalSpec(MicrowaveBaseModel):
+    """Specifies the desired voltage pattern across conductors for a transmission line mode.
+
+    Identifies which conductors should be at positive versus negative voltage for mode selection
+    and ordering in coupled transmission line systems. Conductors can be identified using point
+    coordinates, structure names, or geometric regions for maximum flexibility.
+
+    Notes
+    -----
+    For most use cases, identifying conductors using a single (x, y) coordinate point or a
+    structure name string is recommended. If desired, users may specify geometric regions using
+    arrays of vertices, which are then converted to points, lines, or polygons to identify
+    conductors via intersections.
+
+    Example
+    -------
+    >>> # Recommended: Identify conductors using point coordinates
+    >>> # Differential mode: conductor 1 positive, conductor 2 negative
+    >>> diff_mode = TerminalSpec(
+    ...     plus_terminals=((1.0, 0.5),),  # Point inside positive conductor
+    ...     minus_terminals=((-1.0, 0.5),)  # Point inside negative conductor
+    ... )
+    >>>
+    >>> # Common mode: both conductors positive relative to ground
+    >>> common_mode = TerminalSpec(
+    ...     plus_terminals=((1.0, 0.5), (-1.0, 0.5)),  # Both conductors positive
+    ...     minus_terminals=()  # Ground plane is reference
+    ... )
+    >>>
+    >>> # Alternative: Identify conductors by structure name
+    >>> named_mode = TerminalSpec(
+    ...     plus_terminals=("trace1",),
+    ...     minus_terminals=("trace2",)
+    ... )
+    >>>
+    >>> # Advanced: Identify conductor using polygon region
+    >>> import numpy as np
+    >>> polygon_mode = TerminalSpec(
+    ...     plus_terminals=(np.array([[0, 0], [1, 0], [1, 1], [0, 1]]),),  # Square region
+    ...     minus_terminals=()
+    ... )
+    """
+
+    plus_terminals: tuple[ConductorIdentifierType, ...] = pd.Field(
```

Comment:
Definitely optional, but I think `plus` and `minus` would also be acceptable here instead of  `plus_terminals` and `minus_terminals`